core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -96,7 +96,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (!config.containsMacro("timeTable") && config.timeTable != null) {
       pipelineConfigurer.createDataset(config.timeTable, KeyValueTable.class, DatasetProperties.EMPTY);
     }
-    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    if (config.getSchema() != null) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
```

Comment:
when you set a null output schema at configure time, it means the schema can either be variable or it is not known until runtime. If schema is null and format is not text, you should be setting output schema here to null.